File: 189. Rotate Array.py

class Solution(object):
    def rotate(self, nums, k):

        if k==0:
            return
        if len(nums)==1:
            return
        s=len(nums)        
        sub1=nums[:s-k]

        
        sub2=nums[s-k:]
        
        nums[:]=sub2+sub1

File: 189. Rotate Array.cpp

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;



class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> res=nums;
        
        
        int i,j, len=nums.size();
        if(k>len){
            k=k%len;
        }
        for(i=len-k,j=0;i<len;i++,j++){
            nums[j]=res[i];
        }
         for(i=0;i<len-k;i++){
            nums[j]=res[i];
            j++;
        }
    }
};
